### PR TITLE
Added mutate friendly quat cross mult

### PR DESCRIPTION
**Pull request by shihaocao**
# Added mutate friendly quat cross mult

Fixes # N/A

### Summary of changes
- Added mutate friendly quat cross mult

### Ptest Effects
None, but does require a psim bump for Flight Software, will be incorporated with Attitude Controller update

### Testing
None required, code is just an overload of an existing test

### Constants
None

### Documentation Evidence
Method is self documenting

### CODE REVIEW

File: include/gnc/inl/utilities.inl [hunk 1]
```diff
@@ -30,6 +30,13 @@ constexpr void quat_cross_mult(lin::Vector<T, 4> const &q1, lin::Vector<T, 4> co
   res(3) = q1(3) * q2(3) - lin::dot(qv1, qv2);
 }
 
+template <typename T>
+constexpr void quat_cross_mult(lin::Vector<T, 4> const &q1, lin::Vector<T, 4> const &q2) {
+  lin::Vector<4, T> res;
```
Comment (kylekrol) [inline include/gnc/inl/utilities.inl]: `
```suggestion
  lin::Vector<T, 4> res;
```

File: include/gnc/utilities.hpp [hunk 2]
```diff
@@ -47,6 +47,16 @@ template <typename T>
 constexpr void quat_cross_mult(lin::Vector<T, 4> const &q1,
         lin::Vector<T, 4> const &q2, lin::Vector<T, 4> &res);
 
+/** @fn quat_cross_mult
+ *  @param[in]  q1
+ *  @param[inout]  q2
+ *  Cross multiplies two quaternions. This function essentially performs
+ *  res = q1 x q2. There is no explicit handling of NaNs built into this
```
Comment (kylekrol) [inline include/gnc/utilities.hpp]:
```suggestion
 *  q2 = q1 x q2. There is no explicit handling of NaNs built into this
```

File: include/gnc/inl/utilities.inl [hunk 3]
```diff
@@ -30,6 +30,13 @@ constexpr void quat_cross_mult(lin::Vector<T, 4> const &q1, lin::Vector<T, 4> co
   res(3) = q1(3) * q2(3) - lin::dot(qv1, qv2);
 }
 
+template <typename T>
+constexpr void quat_cross_mult(lin::Vector<T, 4> const &q1, lin::Vector<T, 4> const &q2) {
```
Comment (kylekrol) [inline include/gnc/inl/utilities.inl]:
```suggestion
constexpr void quat_cross_mult(lin::Vector<T, 4> const &q1, lin::Vector<T, 4> &q2) {
```

**Review by kylekrol — CHANGES_REQUESTED** [review]
Couple small things. Actually don't think this will  compile.

File: include/gnc/inl/utilities.inl (hunk 1 above)
Comment (shihaocao) [reply include/gnc/inl/utilities.inl]: Oof, awkward

File: include/gnc/inl/utilities.inl (hunk 3 above)
Comment (shihaocao) [reply include/gnc/inl/utilities.inl]: f

File: include/gnc/inl/utilities.inl [hunk 4]
```diff
@@ -31,6 +31,11 @@ constexpr void quat_cross_mult(lin::Vector<T, 4> const &q1, lin::Vector<T, 4> co
 }
 
 template <typename T>
+constexpr void quat_cross_mult(lin::Vector<T, 4> const &q1, lin::Vector<T, 4> &q2) {
+  lin::Vector<T, 4> res;
+  quat_cross_mult(q1, q2, res);
+  q2 = res;
+
```
Comment (kylekrol) [inline include/gnc/inl/utilities.inl:39]:
```suggestion
}

```

**Review by kylekrol — APPROVED** [review]

**Pull request merged**